src/crush/CrushWrapper.cc

```diff
@@ -1035,6 +1035,15 @@ int CrushWrapper::detach_bucket(CephContext *cct, int item)
   return bucket_weight;
 }
 
+bool CrushWrapper::check_bucket_relation(crush_bucket *f, crush_bucket *tar){
+  if(!f->size)return false;
```

Comment:
also, you might want to take care of your coding style. it is not consistent with the other part of ceph.

Reply:
Thanks,I'll fix it.